### PR TITLE
Docker Builds 

### PR DESCRIPTION
**Pull request by Sushisource**
Done in more or less the same way we did them in the features repo

### CODE REVIEW

File: .github/workflows/all-docker-images.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        type: string
+      py-ver:
+        description: Python SDK ver to build. Skipped if not specified. Must start with v.
+        type: string
+      do-push:
+        description: If set, push the built images to Docker Hub.
+        type: boolean
+        default: false
+  # To support manual triggering:
+  # AFAICT there is no way to avoid repeating all these inputs over and over w/o resorting to templating
```
Comment (bergundy) [inline .github/workflows/all-docker-images.yml:17]: YAML references?

Comment (bergundy) [reply .github/workflows/all-docker-images.yml:17]: But repeating is fine IMHO.

File: .github/workflows/docker-images.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  build-image:
+    name: Build ${{ inputs.lang }} omes docker image
+    runs-on: ubuntu-latest
+    env:
+      REPO_URL: ${{ github.event.pull_request.head.repo.html_url }}
+    steps:
+      - name: Print build information
+        run: 'echo head_ref: "$GITHUB_HEAD_REF", ref: "$GITHUB_REF"'
+
+      - uses: actions/checkout@v2
```
Comment (bergundy) [inline .github/workflows/docker-images.yml:37]: There's a v3 already

File: .github/workflows/docker-images.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      - uses: actions/setup-go@v2
```
Comment (bergundy) [inline .github/workflows/docker-images.yml:39]: There's a v4 already

File: .github/workflows/all-docker-images.yml (hunk 1 above)
Comment (Sushisource) [reply .github/workflows/all-docker-images.yml:17]: They do not work in GHA.

File: .github/workflows/docker-images.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        with:
+          go-version: '^1.20'
+
+      - name: Lint dockerfile
+        run: docker run --rm -i hadolint/hadolint < dockerfiles/${{inputs.lang}}.Dockerfile
+
+      - name: Build docker image
+        run: |
+          go run . build-worker-image --language ${{ inputs.lang }} \
+          --version ${ inputs.sdk-version }} \
```
Comment (bergundy) [inline .github/workflows/docker-images.yml:49]:
```suggestion
          --version ${ inputs.sdk-version }}
```

**Review by bergundy — APPROVED** [review]
Approved with some minor comments.

**Review by cretz — APPROVED** [review]

**Pull request merged**